### PR TITLE
CONN-2159 Create Document Data Submission schemas

### PR DESCRIPTION
**Pull request by ptambellini01**

### CODE REVIEW

File: src/main/resources/schemas/nhinc/common/NhincCommonEntity.xsd [hunk 1]
```diff
@@ -248,9 +248,41 @@
...
+        <xsd:sequence>
+            <xsd:element name="nhinTargetCommunities" type="nccommon:NhinTargetCommunitiesType" minOccurs="0"/>
+            <xsd:element name="RegisterDocumentSetRequest" type="ihe:RegisterDocumentSetRequestType"/>
+            <xsd:element name="url" type="nccommon:UrlInfoType" minOccurs="0" maxOccurs="1"/>
+        </xsd:sequence>
+    </xsd:complexType>
+    <xsd:element name="RespondingGateway_RegisterDocumentSetSecuredRequest" type="tns:RespondingGateway_RegisterDocumentSetSecuredRequestType"/>
 
+    <xsd:complexType name="RespondingGateway_RegisterDocumentSetResponseRequestType">
+        <xsd:sequence>
```
Comment (TabassumJafri) [inline src/main/resources/schemas/nhinc/common/NhincCommonEntity.xsd]: Is this the response Type and not the deferred response request? If yes, can you name it RespondingGateway_RegisterDocumentSetResponseType

File: src/main/resources/schemas/nhinc/common/NhincCommonEntity.xsd [hunk 2]
```diff
@@ -248,9 +248,41 @@
...
+    </xsd:complexType>
+    <xsd:element name="RespondingGateway_RegisterDocumentSetResponseRequest" type="tns:RespondingGateway_RegisterDocumentSetResponseRequestType"/>
 
+    <xsd:complexType name="RespondingGateway_RegisterDocumentSetSecuredResponseRequestType">
+        <xsd:sequence>
+            <xsd:element name="nhinTargetCommunities" type="nccommon:NhinTargetCommunitiesType" minOccurs="0"/>
+            <xsd:element name="RegistryResponse" type="rs:RegistryResponseType"></xsd:element>
+        </xsd:sequence>
+    </xsd:complexType>
+    <xsd:element name="RespondingGateway_RegisterDocumentSetSecuredResponseRequest" type="tns:RespondingGateway_RegisterDocumentSetSecuredResponseRequestType"/>
```
Comment (TabassumJafri) [inline src/main/resources/schemas/nhinc/common/NhincCommonEntity.xsd]: Same comments as above

File: src/main/resources/schemas/nhinc/common/NhincCommonAdapter.xsd [hunk 3]
```diff
@@ -116,6 +116,24 @@
     </xsd:complexType>
     <xsd:element name="RespondingGateway_CrossGatewayProvideAndRegisterDocumentSetRequestResponse" type="tns:RespondingGateway_CrossGatewayProvideAndRegisterDocumentSetRequestResponseType"/>
     <!-- ====================================================================
+    Document Register Schemas.
+    ==================================================================== -->
+    <xsd:complexType name="RespondingGateway_CrossGatewayRegisterDocumentSetRequestRequestType">
```
Comment (TabassumJafri) [inline src/main/resources/schemas/nhinc/common/NhincCommonAdapter.xsd]: Why do we need these?

File: src/main/resources/schemas/nhinc/common/NhincCommonAdapter.xsd [hunk 4]
```diff
@@ -116,6 +116,24 @@
...
+        <xsd:sequence>
+            <xsd:element ref="ihe:RegisterDocumentSetRequest"/>
+            <xsd:element name="assertion" type="nccommon:AssertionType"/>
+        </xsd:sequence>
+    </xsd:complexType>
+    <xsd:element name="RespondingGateway_CrossGatewayRegisterDocumentSetRequestRequest" type="tns:RespondingGateway_CrossGatewayRegisterDocumentSetRequestRequestType"/>
+    <xsd:complexType name="RespondingGateway_CrossGatewayRegisterDocumentSetRequestResponseType">
+        <xsd:sequence>
```
Comment (TabassumJafri) [inline src/main/resources/schemas/nhinc/common/NhincCommonAdapter.xsd]: Why do we need this?

File: src/main/resources/schemas/nhinc/common/NhincCommonAdapter.xsd [hunk 5]
```diff
@@ -116,6 +116,24 @@
...
+            <xsd:element ref="ihe:RegisterDocumentSetRequest"/>
+            <xsd:element name="assertion" type="nccommon:AssertionType"/>
+        </xsd:sequence>
+    </xsd:complexType>
+    <xsd:element name="RespondingGateway_CrossGatewayRegisterDocumentSetRequest" type="tns:RespondingGateway_CrossGatewayRegisterDocumentSetRequestType"/>
+    <xsd:complexType name="RespondingGateway_CrossGatewayRegisterDocumentSetResponseType">
+        <xsd:sequence>
+            <xsd:element ref="rs:RegistryResponse"/>
+            <xsd:element name="assertion" type="nccommon:AssertionType"/>
```
Comment (mhpnguyen) [inline src/main/resources/schemas/nhinc/common/NhincCommonAdapter.xsd]: Question: do we need to have assertion back in response?

File: src/main/resources/schemas/nhinc/common/NhincCommonEntity.xsd [hunk 6]
```diff
@@ -248,9 +248,41 @@
...
+            <xsd:element name="nhinTargetCommunities" type="nccommon:NhinTargetCommunitiesType" minOccurs="0"/>
+            <xsd:element name="RegisterDocumentSetRequest" type="ihe:RegisterDocumentSetRequestType"/>
+            <xsd:element name="url" type="nccommon:UrlInfoType" minOccurs="0" maxOccurs="1"/>
+        </xsd:sequence>
+    </xsd:complexType>
+    <xsd:element name="RespondingGateway_RegisterDocumentSetSecuredRequest" type="tns:RespondingGateway_RegisterDocumentSetSecuredRequestType"/>
 
+    <xsd:complexType name="RespondingGateway_RegisterDocumentSetResponseType">
+        <xsd:sequence>
+            <xsd:element name="assertion" type="nccommon:AssertionType"/>
```
Comment (mhpnguyen) [inline src/main/resources/schemas/nhinc/common/NhincCommonEntity.xsd]: Question: similar like the one before

File: src/main/resources/schemas/nhinc/common/NhincCommonAdapter.xsd [hunk 7]
```diff
@@ -116,6 +116,24 @@
...
+    Document Register Schemas.
+    ==================================================================== -->
+    <xsd:complexType name="RespondingGateway_CrossGatewayRegisterDocumentSetRequestType">
+        <xsd:sequence>
+            <xsd:element ref="ihe:RegisterDocumentSetRequest"/>
+            <xsd:element name="assertion" type="nccommon:AssertionType"/>
+        </xsd:sequence>
+    </xsd:complexType>
+    <xsd:element name="RespondingGateway_CrossGatewayRegisterDocumentSetRequest" type="tns:RespondingGateway_CrossGatewayRegisterDocumentSetRequestType"/>
+    <xsd:complexType name="RespondingGateway_CrossGatewayRegisterDocumentSetResponseType">
```
Comment (TabassumJafri) [inline src/main/resources/schemas/nhinc/common/NhincCommonAdapter.xsd]: We do not need this object -->RespondingGateway_CrossGatewayRegisterDocumentSetResponseType

File: src/main/resources/schemas/nhinc/common/NhincCommonEntity.xsd (hunk 6 above)
Comment (ptambellini01) [reply src/main/resources/schemas/nhinc/common/NhincCommonEntity.xsd]: We do not need the assertion or response.

File: src/main/resources/schemas/nhinc/common/NhincCommonAdapter.xsd (hunk 5 above)
Comment (ptambellini01) [reply src/main/resources/schemas/nhinc/common/NhincCommonAdapter.xsd]: We do not need the assertion or response.

Comment (ptambellini01) [reply src/main/resources/schemas/nhinc/common/NhincCommonAdapter.xsd]: No assertion is necessary and response is already defined  rs.xsd

**Pull request merged**